src/transformers/data/data_collator.py

```diff
@@ -1611,3 +1611,29 @@ def numpy_mask_tokens(self, inputs: Any) -> Tuple[Any, Any, Any, Any]:
             ) & masked_indices[i]
 
         return inputs.astype(np.int64), perm_mask, target_mapping, labels.astype(np.int64)
+
+@dataclass
+class DataCollatorWithFlattening(DefaultDataCollator):
+    return_tensors: str = "pt"
+    def __init__(self, return_position_ids=True):
+        self.return_position_ids=return_position_ids
+        warnings.warn(
+            "Using `DataCollatorForBatchFlattening` will flatten the entire mini batch into single long sequence."
+            "Make sure your attention computation is able to handles it!"
+        )
```

Comment:
I think we keep track of a list of models that support FA2, if needed for guidance here!